src/isc-dhcp/isc-dhcp-4.3.1_dhcrelay-custom-circuit_id-remote_id-and-bridge-iface-support.patch

```diff
@@ -278,12 +364,159 @@ diff -ruN a/isc-dhcp/relay/dhcrelay.c b/isc-dhcp/relay/dhcrelay.c
 +			ip->remote_id = (uint8_t *)remote_id_buf;
 +			ip->remote_id_len = len;
 +
-+			log_debug("sending on %s option82:remote_id='%s'(%d)",
-+					  ip->name, (char *)ip->remote_id, ip->remote_id_len);
++			//log_debug("Sending on %s option82:remote_id='%s' (%d)",
++			//          ip->name, (char *)ip->remote_id, ip->remote_id_len);
 +		}
 +	}
 +
  	/* Sanity check.  Had better not ever happen. */
  	if ((ip->circuit_id_len > 255) ||(ip->circuit_id_len < 1))
  		log_fatal("Circuit ID length %d out of range [1-255] on "
-
+@@ -1102,7 +1324,7 @@
+ 	 * If not, forward without adding the option.
+ 	 */
+ 	if (max - sp >= optlen + 3) {
+-		log_debug("Adding %d-byte relay agent option", optlen + 3);
++		//log_debug("Adding %d-byte relay agent option", optlen + 3);
+ 
+ 		/* Okay, cons up *our* Relay Agent Information option. */
+ 		*sp++ = DHO_DHCP_AGENT_OPTIONS;
+@@ -1726,3 +1948,138 @@
+ 
+ 	exit(0);
+ }
++
++#define MAX_PORT_CONFIG_LINE_LEN 1024
++
++// Allocates and loads global map g_interface_name_alias_map
++// Also sets global g_interface_name_alias_map_size
++static int
++load_interface_alias_map(void) {
++	int i = 0;
++	FILE *fp = NULL;
++	char line[MAX_PORT_CONFIG_LINE_LEN] = { 0 };
++	char port_config_file_path[] = "/usr/share/sonic/hwsku/port_config.ini";
```

Comment:
it is better to give an option to specify which file to load. Also, you do not need to load port_config.ini file, you can use cfggen to generate your own port mapping file and load that port mapping file. In this case, we do not need to parse the port_config.ini, you can rely on cfggen's logic to get the correct port name mapping.

Reply:
I considered that solution, but felt as though writing the output from cfggen to a file and reading that file back in was a bit excessive. However, it does prevent the need for maintaining a separate port_config.ini parser here.

That said, with my current solution, if the format of port_config.ini were to change, this parser would have to be updated. However, if we go with the cfggen solution you suggested and the cfggen output format ever changes, we would need to update the parsing of the cfggen output and the writing of the port mapping file here. I guess the decision comes down to the question: which format is less likely to change over time?

Reply:
Done. Now parsing output from sonic-cfggen and writing to a file in docker_init.sh, then isc-dhcp-relay reads that file at startup.